test/common/testlib.py

```diff
@@ -1749,12 +1749,9 @@ def setUp(self, restrict: bool = True) -> None:
             # First create all machines, wait for them later
             for key in sorted(provision.keys()):
                 options = dict(provision[key])
-                if 'address' in options:
-                    del options['address']
-                if 'dns' in options:
-                    del options['dns']
-                if 'dhcp' in options:
-                    del options['dhcp']
+                options.pop('address', None)
+                options.pop('dns', None)
+                options.pop('dhcp', None)
                 if 'restrict' not in options:
                     options['restrict'] = restrict
```

Comment:
These two adjacent lines look a bit like we might get a future complaint about using `.setdefault()`, no?